### PR TITLE
Factor Signal, Script, Utils modules to ouroboros-network-testing

### PR DESCRIPTION
**Pull request by bolt12**

### CODE REVIEW

**Comment by bolt12** [comment]
Should also probably move `IPv4` and `IPv6` orphan `Arbitrary` instances to `ouroboros-network-testing`

File: ouroboros-network-testing/src/Ouroboros/Network/Testing/Data/AbsBearerInfo.hs [hunk 1]
```diff
@@ -241,6 +241,7 @@ instance Arbitrary AbsBearerInfo where
       | a <- shrink (abiSDUSize abi)
       ]
 
+
```
Comment (coot) [inline ouroboros-network-testing/src/Ouroboros/Network/Testing/Data/AbsBearerInfo.hs:244]: This change seems unrelated to this commit (introduced in `Added github actions/nix testing`).

**Review by coot — APPROVED** [review]
LGTM

File: ouroboros-network-testing/src/Ouroboros/Network/Testing/Data/AbsBearerInfo.hs (hunk 1 above)
Comment (bolt12) [reply ouroboros-network-testing/src/Ouroboros/Network/Testing/Data/AbsBearerInfo.hs:244]: Yeah, I'll remove it, I added it just to trigger hydra to build and not use cache.
